lib/class-wp-duotone-gutenberg.php

```diff
@@ -0,0 +1,375 @@
+<?php
+/**
+ * WP_Duotone_Gutenberg class
+ *
+ * @package gutenberg
+ * @since 6.3.0
+ */
+
+/**
+ * Manages which duotone filters need to be output on the page.
+ *
+ * @access public
+ */
+class WP_Duotone_Gutenberg {
+	/**
+	 * An array of Duotone presets from global, theme, and custom styles.
+	 *
+	 * Example:
+	 * [
+	 *      'blue-orange' =>
+	 *          [
+	 *              'slug'  => 'blue-orange',
+	 *              'colors' => [ '#0000ff', '#ffcc00' ],
+	 *          ]
+	 *      ],
+	 *      …
+	 * ]
+	 *
+	 * @since 6.3.0
+	 * @var array
+	 */
+	protected static $global_styles_presets = array();
```

Comment:
I think `private` is fine. In global styles classes, we made things `protected` so we could extend the class rather than copying things over, but that ended up being messier so we've switched back now.

Reply:
For those playing along at home, https://github.com/WordPress/gutenberg/pull/46579 is the PR switching the `WP_Theme_JSON` override class to being bundled. Following the same approach for this new duotone class sounds good 👍